lib/TfIdfVectorizer.jsm

```diff
@@ -0,0 +1,74 @@
+
+const REGEXP_SPLITS = new RegExp(`[${UNICODE_SPACE}${UNICODE_SYMBOL}${UNICODE_PUNCT}]+`);
+const REGEXP_ALPHANUMS = new RegExp(`^[${UNICODE_NUMBER}${UNICODE_MARK}${UNICODE_LETTER}]+$`);
+
+this.TfIdfVectorizer = class TfIdfVectorizer {
+  /**
+   * Downcases the text, and splits it into consecutive alphanumeric characters.
+   * This is locale aware, and so will not strip accents. This uses "word
+   * breaks", and os is not appropriate for languages without them
+   * (e.g. Chinese).
+   */
+  tokenize(text) {
+    return text.toLocaleLowerCase().split(REGEXP_SPLITS).filter(tok => tok.match(REGEXP_ALPHANUMS));
```

Comment:
These two pieces of regex processing could be quite CPU intensive, we might want to keep an eye on it by profiling.